components/calculation/MeasureHighlightingPanel.tsx

```diff
@@ -32,28 +27,35 @@ export default function MeasureHighlightingPanel({ patientId }: MeasureHighlight
   const { classes } = useStyles();
   const [searchValue, setSearchValue] = useState('');
   const detailedResultLookup = useRecoilValue(detailedResultLookupState);
-  const { parsedHTML, defIds } = useMemo(() => {
-    const defIds: Record<string, string> = {};
+  const parsedHTML = useMemo(() => {
     const parsedHTML = parse(detailedResultLookup[patientId]?.detailedResults?.[0].html || '', {
       replace: elem => {
-        if (elem.type === 'text') {
-          const data = (elem as DomText).data;
-          if (data.match(expressionDefRegex)) {
-            const splitData = data.split('"');
-            defIds[splitData[1]] = splitData[1].toLowerCase().replace(' ', '-');
-            return <span id={defIds[splitData[1]]}>{data}</span>;
-          }
+        if ((elem as DomElement).attribs?.['data-statement-name']) {
+          const statementName = (elem as DomElement).attribs?.['data-statement-name'];
+          const statementResult = detailedResultLookup[patientId]?.detailedResults?.[0].statementResults.find(
+            statement => statement.statementName === statementName
```

Comment:
need to also check the value of `attribs['data-library-name']` and use that in the `.find` for a statement result since that will guarantee the uniqueness. Just statement name won't